src/scorepy/events.hpp

```diff
@@ -16,9 +17,17 @@ inline const std::string& make_region_name(const char* module_name, const char*
     return region;
 }
 
-void region_begin(const std::string& region_name, std::string module, std::string file_name,
-                  std::uint64_t line_number);
-void region_end(const std::string& region_name);
+void region_begin(const std::string& function_name, const std::string& module,
+                  const std::string& file_name, const std::uint64_t line_number,
+                  const std::uintptr_t& identifier);
+void region_begin(const std::string& function_name, const std::string& module,
+                  const std::string& file_name, const std::uint64_t line_number);
+
+void region_end(const std::string& function_name, const std::string& module,
+                const std::uintptr_t& identifier);
+void region_end(const std::string& function_name, const std::string& module);
+
+void region_end_error_handling(const std::string& region_name);
```

Comment:
I wouldn't list this here as it is internal to this file (only used for error handling the the *_end functions) and not really meant to be called from outside

Reply:
I would. And it won't hurt.